eq-author/src/App/questionConfirmation/Design/index.js

```diff
@@ -27,14 +29,31 @@ export class UnwrappedQuestionConfirmationRoute extends React.Component {
     onDeleteQuestionConfirmation: PropTypes.func.isRequired,
   };
 
-  handleDeleteClick = () => {
-    this.props.onDeleteQuestionConfirmation(
-      this.props.data.questionConfirmation
+  state = {
+    showDeleteConfirmDialog: false,
+  };
+
+  handleOpenDeleteConfirmDialog = () =>
+    this.setState({ showDeleteConfirmDialog: true });
+
+  handleCloseDeleteConfirmDialog = cb =>
+    this.setState(
+      { showDeleteConfirmDialog: false },
+      isFunction(cb) ? cb : null
```

Comment:
I think `setState` will handle this function check for us. I don't think we need to do it ourselves.

Reply:
`cb` isn't always a function and `setState` will throw `Error: Invalid argument passed as callback. Expected a function. Instead received: ...` when an invalid type is passed as a callback.

Reply:
I was expecting to be `undefined` is that not the case?

Reply:
No longer applicable.